lib/meilisearch-rails.rb

```diff
@@ -306,7 +306,7 @@ def settings(*args)
         SafeIndex.log_or_throw(:settings, @raise_on_failure) do
           @index.settings(*args)
         rescue ::MeiliSearch::ApiError => e
-          return {} if e.code == 404 # not fatal
+          return {} if e.code == 'index_not_found' # not fatal
```

Comment:
I'm not sure why we had the 404 in the first place, but `index_not_found` is just one issue that could happen, shouldn't we make it more generic? Like parsing the error response to grab a "match-all" case?